### ISSUE TITLE
Mangaplus: Add a true chapter number for downloads

### ISSUE THREAD

**Issue by Antsy4904** [keiyoushi/tachiyomi-extensions-source#5056]
### Source name

Mangaplus

### Source language

English

### Describe your suggested feature

I noticed that you added a setting to remove redundant chapter numbers, that got me thinking about a possible setting to add chapter numbers when there's none. 

Lots of manga on mangaplus has multiple special chapters that are all named `Ex - Bonus Chapter` and this causes issues when you download chapters because the app uses that as the name and can't tell them apart. If you download one it thinks you downloaded them all. If you're just reading without downloading then it can tell them apart though. 

You can see this in action  right now if you read and download Existential Unplugged. 

I would love a setting that added some kind of "true" number or unique ID to each chapter so that they would get unique filename no matter what they are named. 

### Other details

I hope this is within the scope of the extensions, I've only had the issue with mangaplus.

PS this template needs to be updated so it asks for 0.16.5. 

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/mihonapp/mihon/issues/new/choose).
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by TheUnlocked** [comment]
FYI: https://github.com/mihonapp/mihon/pull/2305

**Comment by Antsy4904** [comment]
Yeah, this is fixed with this: https://github.com/mihonapp/mihon/commit/58b25d697f7987e9888344e815d5646ec010a663

That was my reason to change how chapter numbers worked so I'm closing this.
